---
title: "ATAC-seq data quality control using the ATACseqQC package"
author: "Haibo Liu, Jianhong Ou, Lihua Julie Zhu"
date: "`r format(Sys.time(), '%B %d, %Y')`"
package: "`r packageVersion('ATACseqQCWorkshop')`"
bibliography:  "`r system.file('vignettes', 'ATAC-seq.bib', package = 'ATACseqQCWorkshop')`"
csl:  "`r system.file('vignettes', 'nature.csl', package = 'ATACseqQCWorkshop')`"
vignette:  >
  %\VignetteIndexEntry{ATACseqQCWorshop}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
output: 
  html_document:
    theme: simplex
    toc: true
    toc_float: true
    toc_depth: 3
    fig_caption: true
---

```{r loadLibrariesNSetOpts, include = FALSE}
suppressPackageStartupMessages({
  library(ATACseqQC)
  library(ChIPpeakAnno)
  library(BSgenome.Hsapiens.UCSC.hg19)
  library(TxDb.Hsapiens.UCSC.hg19.knownGene)
  library(MotifDb)
  library(SRAdb)
  library(GenomicAlignments)
})
knitr::opts_chunk$set(echo = TRUE, eval=TRUE, fig.width=8, 
                      fig.height=6, warning = FALSE, 
                      message = FALSE, encoding = "UTF-16")
```

## Instructors and contact information
***
* [Haibo Liu](https://github.com/haibol2016) (Department of Animal Science, Iowa State University, Ames, IA, 50010, USA. haibol@iastate.edu)  
* [Jianhong Ou](https://github.com/jianhong) (Department of Cell Biology, Duke University Medical Center, Durham, NC, 27710, USA. Jianhong.ou@duke.edu)   
* [Lihua Julie Zhu](https://github.com/LihuaJulieZhu) (Corresponding author, Department of Molecular, Cell and Cancer Biology, Program in Molecular Medicine, Program in Bioinformatics and Integrative Biology, Worcester, MA, 01655, USA. Julie.Zhu@umassmed.edu)  


## Introduction to the ATAC-seq technology
***
In eukaryotes, nucleosomes are basic DNA packaging units, each of which consists of a nucleosome core particle (NPC), linker DNA and linker histone H1. The NPC is composed of ~147 bp of DNA wrapped around a histone core octamer, spaced from adjacent NPCs by a linker DNA of ~20-90 bp. Linker histones binds additional 20 bp of linker DNA at its entry/exit points to a NPC to stabilize nucleosome conformation and higher-order chromatin assembly. Nucleosomes can be hierarchically assembled into higher-order structures, eventually into chromatin or even chromosomes [@Pederson1986; @Bednar2017] (Figure 1, adapted from [Nature Education 1(1):26](https://www.nature.com/scitable/topicpage/dna-packaging-nucleosomes-and-chromatin-310)). In cells, different genomic regions are packaged with different accessibility to transcriptional machinery. Most notably, promoters and enhancers of actively transcribed genes are devoid of histone interaction, which are called "open" chromatin regions (Figure 2, adapted from Wang _et al_ [@Wang2012]). Thus, the interplay between histones and DNA serves as an important layer for controlling gene expression [@Gilbert2005; @Perino2016]. Therefore, it is important to determine the chromatin accessibility to better understand gene expression regulation in cells. 
<br/>
```{r, out.width ='80%', out.height ='80%', fig.align='center', fig.cap='Figure 1. DNA packaging in eukaryotic cells.', echo = FALSE}
knitr::include_graphics(system.file('vignettes', 'Figure 1. DNA packaging.png', package = 'ATACseqQCWorkshop'))
```
<br/>
```{r, out.width='80%', fig.align='center', fig.cap='Figure 2. Open and closed chromatin.', echo = FALSE}
knitr::include_graphics(system.file('vignettes', 'Figure 2. Open and closed chromatin.png', package = 'ATACseqQCWorkshop'))
```
<br/>
In recent years, a few methods have been developed to profile genome-wide chromatin accessibility (Figure 3, adapted from Tsompana Tsompana and Buck [@Tsompana2014]) For review, see Tsompana and Buck [@Tsompana2014]. Among these methods, ATAC-seq (an assay for Transposase-Accessible Chromatin using sequencing) is a rapid and sensitive method for profiling chromatin accessibility [@Buenrostro2013]. Compared to other methods, such as MNase-seq, FAIRE-seq and DNase-seq, ATAC-seq allows comparable or even higher signal-to-noise ratio, but requires much less amount of the biological materials and time to process. Briefly, hyperactive Tn5 transposases preloaded with adapters are first added to simultaneously tag and fragment open chromatin in nuclei (500~50,000) isolated from fresh or cryopreserved tissues/cells at nearly native states, a process called tagmentation. The tagged DNA fragments are then amplified and simultaneously sequencing adapters and indices compatible to Illumina sequencing platforms are added by using PCR with optimized cycles (Figure 4, adapted from Buenrostro _et al_ [@Buenrostro2013]).  
<br/>
```{r, out.width='80%', fig.align='center', fig.cap='Figure 3. Signal features generated by different methods for profiling chromatin accessibility.', echo = FALSE}
knitr::include_graphics(system.file('vignettes', 'Figure 3. Chromatin accessibility.png', package = 'ATACseqQCWorkshop'))
```
<br/>

```{r, out.width='80%', fig.align='center', fig.cap='Figure 4. Schematic showing of ATAC-seq library preparation.', echo = FALSE}
knitr::include_graphics(system.file('vignettes', 'Figure 4. library preparation.png', package = 'ATACseqQCWorkshop'))
```
<br/>
Since its first development, ATAC-seq has been used for quite a few purposes [@Buenrostro2013; @Buenrostro2015b]:  

* Infer nucleosome positioning  
* Profile open chromatin regions  
* Identify transcription factor (TF) footprints  
* Infer gene transcriptional regulation  
* Determine cell state/identity in combination with single cell technologies  

The ATAC-seq wet-lab protocol is relatively simple and easy. However, to ensure a successful ATAC-seq experiment, step-by-step quality assurance processes, including both wet lab quality control and _in silico_ quality assessment, are essential (Figure 5, modified from Milani _et al_ [@Milani2016]). In this workshop we will mainly cover the post-alignment QC steps using our ATACseqQC package [@Ou2018].  
<br/>
```{r, out.width='80%', fig.align='center', fig.cap='Figure 5. Quality control steps in ATAC-seq assays.', echo = FALSE}
knitr::include_graphics(system.file('vignettes', 'Figure 5. QC steps in ATAC-seq.png', package = 'ATACseqQCWorkshop'))
```
<br/>

## Preprocessing of ATAC-seq data
***
Before using the ATACSeqQC package to assess the quality of ATAC-seq data, several preprocessing steps are needed as follows. These steps are usually performed in UNIX environments using open source software other than R. Detailed information about how to preprocess ATAC-seq data is provided in **"vignettes/Preprocessing.scripts.for.BioC2019.tutorial.txt"**.  

* Check quality of raw reads using [FASTQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/).  
* Generate an interactive summary report as a HTML file using MultiQC [@Ewels2016].  
* Trim sequencing adapters using Trimmomatic [@Bolger2014] or [Trim Galore{!}](http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/).  
* Map the reads to the reference genome of choice using Bowtie/Bowtie2 [@Langmead2009; @Langmead2012] or BWA-mem [@Li2009a].  
* Convert and sort alignment files using SAMtools [@Li2009b] or sambamba [@Tarasov2015].  
* Filter aligned reads  
  + Remove reads derived from organelles using SAMtools [@Li2009b]  
  + Remove duplicates using [picard tools](https://broadinstitute.github.io/picard/)  
  + Remove mapping artifacts (fragments < 38 bp [@Tsompana2014] and fragments > 2 kb, reads with un-coordinated mapping) using custom scripts  

## ATACseqQC workflow
***
In this workshop we will start with preprocessed alignment files in the BAM formats and use the R/BioConductor package ATACseqQC to perform some representative steps of post-alignment QC. 

```{r setup}
## prepare the example BAM files for importing
bamFile <- system.file("extdata", "GL1.bam",
                        package="ATACseqQC", mustWork=TRUE)
bamFileLabels <- gsub(".bam", "", basename(bamFile))
```

### Assessing ATAC-seq read mapping status
***
FASTQC can generate a summary of the quality of raw reads, but it can't reveal the quality of post-aligned reads or mapping status, such as mapping rate, duplicate rate, genome-wide distribution, mapping quality, and contamination. The `bamQC` function can serve these needs, which requires sorted BAM files with duplicate reads marked as input. 

```{r bamQC}
## bamQC
bamQC(bamFile, outPath=NULL)
```

### Assessing insert size distribution
***
It is known that in ATAC-seq experiments, tagmentation of Tn5 transposases produce signature size pattern of fragments derived from nucleosome-free regions, mononucleosome, dinucleosome, trinucleosome and longer oligonucleosome from other open chromatin regions (Figure 6, adapted from Li _et al_ [@Li2019]) . A typical distribution of insert fragment size is shown in (Figure 7, adapted from Buenrostro _et al_ [@Buenrostro2015a]. The function `fragSizeDist` in the ATACseqQC package can be used to generate such a distribution plot. Please note the pre-filtered BAM files need to be used to get an unbiased distribution of insert fragment size in the ATAC-seq library.  
<br/>
```{r, out.width='80%', fig.align='center', fig.cap='Figure 6. Fragments generated by Tn5 tagementation in ATAC-seq.', echo = FALSE}
knitr::include_graphics(system.file('vignettes', 'Figure 6. Fragments generated by Tn5 tagmentation in ATAC-seq.png', package = 'ATACseqQCWorkshop'))
```
<br/>
```{r, out.width='80%', fig.align='center', fig.cap='Figure 7. Library insert fragment size distribution in a typical ATAC-seq assay.', echo = FALSE}
knitr::include_graphics(system.file('vignettes', 'Figure 7. Library insert size distribution.png', package = 'ATACseqQCWorkshop'))
```
<br/>
```{r fragSizeDist, fig.align='center', fig.height=6, fig.width=8, fig.cap = "Figure 8. Size distribution of ATAC-seq library insert fragments."}
fragSize <- fragSizeDist(bamFile, bamFileLabels)
```
<br/>  

### Shifting aligned reads  
***
Tagmentation of Tn5 transposases produce 5' overhang of 9 base long, the coordinates of reads mapping to the positive and negative strands need to be shifted by + 4 and - 5, respectively [@Nag1985; @Buenrostro2013]. The functions `shiftGAlignmentsList` and `shiftReads` in the ATACseqQC package can be used for this purpose.

```{r shiftAlignment}
## bamfile tags to be read in
possibleTag <- combn(LETTERS, 2)
possibleTag <- c(paste0(possibleTag[1, ], possibleTag[2, ]),
                 paste0(possibleTag[2, ], possibleTag[1, ]))
bamTop100 <- scanBam(BamFile(bamFile, yieldSize = 100),
                     param = ScanBamParam(tag=possibleTag))[[1]]$tag
tags <- names(bamTop100)[lengths(bamTop100)>0]

## files will be output into outPath
outPath <- "splitBam"
if (dir.exists(outPath))
{
  unlink(outPath, recursive = TRUE, force = TRUE)
}
dir.create(outPath)
## shift the coordinates of 5'ends of alignments in the bam file
seqlev <- "chr1" ## subsample data for quick run
which <- as(seqinfo(Hsapiens)[seqlev], "GRanges")
gal <- readBamFile(bamFile, tag=tags, which=which, 
                   asMates=TRUE, bigFile=TRUE)
shiftedBamFile <- file.path(outPath, "shifted.bam")
gal1 <- shiftGAlignmentsList(gal, outbam=shiftedBamFile)

```

### Splitting BAM files
***
The shifted reads need to be split into different bins, namely bins for reads from nucleosome-free regions, and regions occupied by mononucleosome, dinucleosomes, and trinucleosomes. Shifted reads that do not fit into any of the three bins could be discarded. The function `splitGAlignmentsByCut` has been implemented to meet this demand.

```{r splitBAM}
## run program for chromosome 1 only
txs <- transcripts(TxDb.Hsapiens.UCSC.hg19.knownGene)
txs <- txs[seqnames(txs) %in% "chr1"]
genome <- Hsapiens
## split the reads into bins for reads derived from nucleosome-free regions, mononucleosome, dinucleosome and trinucleosome. And save the binned alignments into bam files.
objs <- splitGAlignmentsByCut(gal1, txs=txs, genome=genome, outPath = outPath)
## list the files generated by splitGAlignmentsByCut.
dir(outPath)
```

### Plotting global signal distribution around transcriptional start sites (TSSs)
***
Previous studies have shown that transcriptionally active elements, such as promoters and enhancers, are defined by short regions of DNA that are devoid of direct histone interactions. These regions of "open" chromatin are usually occupied by transcription factors that facilitate gene transcription. By contrast, the promoters of genes that are not actively expressed in a given cell type exhibit much tighter association with histones, which prevents transcription factors from activating transcription and contributes to gene repression. Typical nucleosome density distributions around TSSs of actively transcribed genes and active enhancers are shown in Figure 9, adapted from Baldi [@Baldi2018]. The function `enrichedFragments` is for calculating signals around TSSs using the split BAM objects. And the function `featureAlignedHeatmap` is for generating a heatmap showing signal distribution around TSSs. The function `matplot` is for summarizing the signal distribution around TSSs using a density plot.  
<br/>
```{r, out.width='80%', fig.align='center', fig.cap='Figure 9. Nucleosome density distribution around TSSs and TF binding sites.', echo = FALSE}
knitr::include_graphics(system.file('vignettes', 'Figure 8. Nucleosome density distribution around TSS and TF binding site.png', package = 'ATACseqQCWorkshop'))
```
<br/>
```{r heatmap, fig.height=6, fig.width=4, fig.align='center', fig.cap = "Figure 10. Heatmap showing distributions of reads from different bins around TSSs."}
bamFiles <- file.path(outPath,
                     c("NucleosomeFree.bam",
                     "mononucleosome.bam",
                     "dinucleosome.bam",
                     "trinucleosome.bam"))

TSS <- promoters(txs, upstream=0, downstream=1)
TSS <- unique(TSS)
## estimate the library size for normalization
(librarySize <- estLibSize(bamFiles))
## calculate the signals around TSSs.
NTILE <- 101
dws <- ups <- 1010
sigs <- enrichedFragments(gal=objs[c("NucleosomeFree", 
                                     "mononucleosome",
                                     "dinucleosome",
                                     "trinucleosome")], 
                          TSS=TSS,
                          librarySize=librarySize,
                          seqlev=seqlev,
                          TSS.filter=0.5,
                          n.tile = NTILE,
                          upstream = ups,
                          downstream = dws)
## log2 transformed signals
sigs.log2 <- lapply(sigs, function(.ele) log2(.ele+1))
#plot heatmap
featureAlignedHeatmap(sigs.log2, reCenterPeaks(TSS, width=ups+dws),
                      zeroAt=.5, n.tile=NTILE)
```

```{r fig.show='hide'}
## get signals normalized for nucleosome-free and nucleosome-bound regions.
out <- featureAlignedDistribution(sigs, 
                                  reCenterPeaks(TSS, width=ups+dws),
                                  zeroAt=.5, n.tile=NTILE, type="l", 
                                  ylab="Averaged coverage")
```

```{r fig.height=4, fig.width=6,  fig.align='center', fig.cap = "Figure 11. Density plot showing distributions of reads from different bins around TSSs. Black line, signal generated from nucleosome-free bin; red line, singal from mononucleosome bin."}
## rescale the nucleosome-free and nucleosome signals to 0~1
range01 <- function(x){(x-min(x))/(max(x)-min(x))}
out <- apply(out, 2, range01)
matplot(out, type="l", xaxt="n", 
        xlab="Position (bp)", 
        ylab="Fraction of signal")
axis(1, at=seq(0, 100, by=10)+1, 
     labels=c("-1K", seq(-800, 800, by=200), "1K"), las=2)
abline(v=seq(0, 100, by=10)+1, lty=2, col="gray")
```

### Streamlining IGV snapshots showing signal distribution along housekeeping genes
***
Housekeeping genes are relatively stably expressed across tissues. Signal enrichment is expected in the regulatory regions of housekeeping genes in successful ATAC-seq experiments, which provides valuable insights into the quality of the ATAC-seq library. We have implemented the function `IGVSnapshot` is to facilitate automatic visualization of signal distribution along any genomic region of interest. CAUTION: no white spaces are allowed in the full path to BAM files.

```{r IGVSnapshot, eval=FALSE, echo = TRUE}
source(system.file("extdata", "IGVSnapshot.R", package = "ATACseqQC"))
IGVSnapshot(maxMem="lm", genomeBuild = "hg19", 
            bamFileFullPathOrURLs = bamFile, 
            geneNames = c("PQLC2", "MINOS1"))
```

### Assessing footprints of DNA-binding factors
***
Genomic regions bound by transcription factors/insulators are locally protected from Tn5 transposase tagmentation and the pattern of Tn5 transposase cutting sites around TF-binding sites can be used to infer the footprints of TFs. The function `factorFootprints` is for plotting footprints of DNA-binding factors.

```{r CTCF_footprints, fig.height=4, fig.width=6,  fig.align='center', fig.cap = "Figure 12. Aggregated CTCF footprints."}
## foot prints
CTCF <- query(MotifDb, c("CTCF"))
CTCF <- as.list(CTCF)
print(CTCF[[1]], digits=2)
sigs <- factorFootprints(shiftedBamFile, pfm=CTCF[[1]], 
                         genome=genome,
                         min.score="90%", seqlev=seqlev,
                         upstream=100, downstream=100)
```

### Assessing sequencing depth and library complexity
***
Although all the above assessments can be used to determine the quality of the available ATAC-seq data, they can't tell whether the sequencing depth is saturated or not, nor whether the library is complex enough for deeper sequencing. The functions `saturationPlot` and `estimateLibComplexity` are for these purposes. CAUTION: only BAM files without removing duplicates are informative for estimating library complexity. 

```{r libComplexity,  fig.height=4, fig.width=6,  fig.align='center', fig.cap = "Figure 13. Scatter plot showing estmated ATAC-seq library complexity."}
estimateLibComplexity(readsDupFreq(bamFile))
```

### Assessing similarity of replicates 
***
If multiple ATAC-seq assays have been performed, signal correlation among replicates can be checked using the function `plotCorrelation`.  

```{r sampleCorrelation,fig.height=4, fig.width=4,  fig.align='center', fig.cap = "Figure 14. Heatmap showing replicate similarity."}
path <- system.file("extdata", package="ATACseqQC", mustWork=TRUE)
bamFiles <- dir(path, "*.bam$", full.name=TRUE)
gals <- lapply(bamFiles, function(bamfile){
               readBamFile(bamFile=bamfile, tag=character(0), 
                          which=GRanges("chr1", IRanges(1, 1e6)), 
                          asMates=FALSE)
         })
txs <- transcripts(TxDb.Hsapiens.UCSC.hg19.knownGene)
plotCorrelation(GAlignmentsList(gals), txs, seqlev="chr1")
```  
<br/>

### Assessing other QC metrics    
***
A few new QC functions have been added to the ATACseqQC package.  

* `distanceDyad`: calculate the distance of potential nucleosome dyad and the linear model for V.  
* `NFRscore`: calculate the ratio of cutting signal immediately adjacent to TSSs and that located to the regions flanking TSSs.  
* `PTscore`: calculate the ratio of read coverage over promoters to read coverage over transcript body.  
* `TSSEscore`: calculate aggregated distribution of reads centered on TSSs and that of reads flanking the corresponding TSSs.  
* `vPlot`: aggregate ATAC-seq Fragment Midpoint vs. Length for a given motif generated over binding sites within the genome.  

## Exercises and live demonstration
***
ATAC-seq datasets from two different studies were downloaded from the ENA Sequence Read Archive (SRA). SRR891269 and SRR891270 are ATAC-seq data for two biological replicates of 50K cells from EBV-transformed lymphoblastoid cell line GM12878 [@Buenrostro2013]. SRR5800801 and SRR5800802 are ATAC-seq data for two replicates of 75k cells from a breast cancer cell line T47 (Valles and Izquierd-Bouldstridge, unpublished). The first dataset is of good quality, while the second is from failed ATAC-seq assays. After raw read quality QC using FASTQC, reads were aligned to the human reference genome **hg38** from the [UCSC Genome Browser](http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz) after removing the alternative sequences (chr_alt chromosomes). The resulting BAM files were further preprocessed as above and are available from [the Google Drive](https://drive.google.com/drive/folders/1zGvqnXAIWelgVk9Gw1Kx5ySlCg0iz7AJ?usp=sharing).  

Two subsets of filtered BAM files from SRR891270, containing alignments on **chr20** only, are provided in this workshop package for quick in-class practice.  

Due to the tight schedule, we will not demonstrate how to perform the QC analyses using the functions `saturationPlot`,`factorFootprints`, `NFRscore`, `PTscore`, `TSSEscore`, or `vPlot`. You may play with the other data and these functions after class. You are encouraged to post your questions at https://support.bioconductor.org with post title "ATACseqQC". 

For your reference, example R scripts for Tasks 1-5 are available at **"inst/vignettes/BioC2019.demo.ATACseqQC.R"**.  

<br/>
First, load the required packages.
```{r loadLibraries, include = TRUE, eval = FALSE}
suppressPackageStartupMessages({
  library(ATACseqQC)
  library(ChIPpeakAnno)
  library(BSgenome.Hsapiens.UCSC.hg38)
  library(TxDb.Hsapiens.UCSC.hg38.knownGene)
  library(MotifDb)
  library(SRAdb)
  library(GenomicAlignments)
})
```

Task 1. Check read alignment statistics using the function`bamQC`.  
To get the location of the BAM file for Tasks 1 and 2, , please copy and run the following scripts in a R console.
```{r BAMFileWithDuplicates4ch20}
dupBamFile <- system.file("vignettes/extdata", 
                       "SRR891270.chr20.full.bam", 
                       package = "ATACseqQCWorkshop")
```

Task 2. Check library complexity for SRR891270 using the function `estimateLibComplexity`. CAUTION: duplicated should NOT be removed from the input BAM file.


Using the following code to get access to the BAM file for Tasks 2-4.  
```{r getBAMFile}
bamFile <- system.file("vignettes/extdata", 
                       "SRR891270.chr20.rmdup.bam", 
                       package = "ATACseqQCWorkshop")
bamFileLabels <- gsub(".rmdup.bam", "", 
                      basename(bamFile))
```

Task 3. Plot the distribution of library insert size using the function `fragSizeDist`.

Task 4. Plot global signal enrichment around TSSs using the function `enrichedFragments`. Please shift, and split the BAM file first. Make sure set seqlev as **chr20**.

Task 5. Visualize signal distribution along genomic regions of genes "SPTLC3", "CRLS1", and "NELFCD", using the function `IGVSnapshot` and the 4 subsets of filtered BAM files.

Download data to a location without white spaces in the full path from the folder **subsetted_filtered_BAM** in the Google Drive, by following **[the shared link](https://drive.google.com/drive/folders/1zGvqnXAIWelgVk9Gw1Kx5ySlCg0iz7AJ?usp=sharing)**.

```{r getFunction}
source(system.file("extdata", "IGVSnapshot.R", package = "ATACseqQC"))
```

## Best practices of ATAC-seq and ATAC-seq data QC
***
A few best practices for ATAC-seq assays are suggested as follows:  

* Digest away background DNA (medium/dead cells) using DNase [@Corces2017]  
* Use fresh/cyropreserved cells/tissues to isolate nuclei [@Buenrostro2013; @Milani2016]  
* Reduce mitochondrial/chloroplast DNA contamination as much as possible [@Corces2017; @Lu2016; @Rickner2019; @Montefiori2017]  
* Optimize the ratio of the amount of Tn5 enzyme to the number of nuclei  
* Optimize the number of PCR cycles [@Buenrostro2015a]  
* Perform Paired-end (PE) sequencing, e.g., 2 x 50bp  
* Sequence > 50M PE reads (~200 M for footprint discovery) [@Buenrostro2013]  

A few best practices for ATAC-seq data analysis are suggested as follows:  

* Perform raw read QC using [FASTQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) before alignment   
* Perform post-alignment QC using ATACSeqQC [@Ou2018]  
* Perform peak calling using a peak caller, such as [MACS2](https://github.com/taoliu/MACS) [@Zhang2008] with non-broadPeak mode or HMMRATAC [@Tarbell2019]. 
* Perform post-peak calling QC   
  + Annotate peaks to nearest genes and generate peak distribution among genomic features using ChIPpeakAnno [@Zhu2010] 
  + Obtain functions of genes associated to peaks using the Genomic Regions Enrichment of Annotations Tool (GREAT) [@McLean2010]  

## Session information
***
```{r sessionInfo, echo = FALSE}
sessionInfo()
```

## References
***
